compositor_pipeline/src/pipeline/decoder/audio/opus.rs

```diff
@@ -49,25 +50,99 @@ impl OpusDecoder {
         )
         .into()
     }
+
+    fn set_end_pts(&mut self, decoded_samples: &DecodedSamples) {
+        let samples_len = decoded_samples.samples.get_number_of_samples();
+        let sample_rate = decoded_samples.sample_rate;
+
+        let chunk_duration = Duration::from_secs_f64(samples_len as f64 / sample_rate as f64);
+        self.last_decoded_pts = Some(decoded_samples.start_pts + chunk_duration);
+    }
+
+    fn should_use_fec(&self, stream_gap: Duration) -> bool {
+        stream_gap > Duration::from_millis(1)
+    }
+
+    fn calculate_stream_gap(&mut self, current_start: Duration) -> Duration {
+        let stream_gap = current_start - *self.last_decoded_pts.get_or_insert(current_start);
+        trace!(
+            "[opus decoder] Calculated stream gap: {} s",
+            stream_gap.as_secs_f64(),
+        );
```

Comment:
```suggestion
        trace!("[opus decoder] Calculated stream gap: {stream_gap:?}");
```
or
```suggestion
        trace!(?stream_gap, "[opus decoder] Detected stream gap}");
```

Reply:
Also, instead of "[opus decoder]", you can check out how to use spans, but I don't think it's necessary here, it's just for future reference